actions_test.go

```diff
@@ -0,0 +1,160 @@
+package opslevel_test
+
+import (
+	ol "github.com/opslevel/opslevel-go/v2023"
+	"github.com/rocktavious/autopilot/v2022"
+	"testing"
+)
+
+func TestWebhookActionHeadersMarshalling(t *testing.T) {
+	// Arrange
+	unstructured := "Accept=\"application/vnd.github+json\"\nAuthorization=\"Bearer XXX\"\nContent-Type=\"application/json\"\nHost=\"app.opslevel.com\"\nOrigin=\"api.opslevel.com\""
+	structured := map[string]string{
+		"Host":          "app.opslevel.com",
+		"Origin":        "api.opslevel.com",
+		"Content-Type":  "application/json",
+		"Accept":        "application/vnd.github+json",
+		"Authorization": "Bearer XXX",
+	}
+	// Act
+	marshalled := ol.CustomActionsToHeaders(unstructured)
+	unmarshalled := ol.CustomActionsFromHeaders(structured)
+	// Assert
+	autopilot.Equals(t, structured, marshalled)
+	autopilot.Equals(t, unstructured, unmarshalled)
+}
+
+func TestCreateWebhookAction(t *testing.T) {
+	//Arrange
+	request := `{"query": 
```

Comment:
I created a new approach to test data that should hopefully reduct the amount of files we have to generate

I also then tried to use the templates in the "right" places to reduce the duplication